Applications/MainApplication/Gui/MainWindow.cpp

```diff
@@ -576,6 +568,12 @@ namespace Ra
                                              QItemSelectionModel::ClearAndSelect | QItemSelectionModel::Current);
     }
 
+    void Gui::MainWindow::onGLInitialized()
+    {
+        // set renderers once OpenGL is configured
+        addRenderer("Forward Renderer", new Engine::ForwardRenderer());

```

Comment:
TODO : Use shared_ptr to share the ownership of renderers between mainapp and viewer